File: solutions/2216-delete-the-middle-node-of-a-linked-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        if (head == null || head.next == null) {
            return null;
        }
        ListNode curr=head;
        int size=0;
        while (curr!=null) {
            size++;
            curr=curr.next;
        }
        int middleIndex=size/2;
        curr=head;
        int count=0;
        while(count<middleIndex-1) {
            count++;
            curr=curr.next;
        }
        if(curr!=null && curr.next!=null) {
            curr.next=curr.next.next;
        }
        return head;
    }
}

File: solutions/2216-delete-the-middle-node-of-a-linked-list/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        
        int l = 0;

        ListNode* temp = head;

        while(temp != NULL) {
            temp = temp->next;
            l++;
        }

        if(l == 1) return NULL;

        int mid = l/2;

        temp = head;

        for(int i = 1; i < mid; i++) {
            temp = temp->next;
        }

        temp->next = temp->next->next;

        return head;

    }
};
